projects/real_world_ovmm/evalai_env.py

```diff
@@ -0,0 +1,303 @@
+
+        if done:
+            metrics = self._extract_scalars_from_info(hab_info)
+            metrics_at_episode_end = {"END." + k: v for k, v in metrics.items()}
+            self._current_episode_metrics = {
+                **metrics_at_episode_end,
+                **self._current_episode_metrics,
+            }
+            if "goal_name" in info:
+                self._current_episode_metrics["goal_name"] = info["goal_name"]
+
+            self._episode_metrics[self._current_episode_key] = self._current_episode_metrics
+            self._current_episode_metrics = {}
+
+        observations = self._env.get_observation()
+        hab_info = None ## not sure what hab_info should be for real world
+        return evaluation_pb2.Package(SerializedEntity=grpc_dumps((observations, done, hab_info)))
+
+    def evalai_update_submission(self, request, context):
+        if len(self._episode_metrics) != self._env_number_of_episodes:
+            raise ValueError(
+                "Not all episodes have been evaluated: "
+                f"number of episodes evaluated ({len(self._episode_metrics)}) != total number of episodes ({self._env_number_of_episodes})."
+            )
+        
+        average_metrics = self._summarize_metrics(self._episode_metrics)
+        self._print_summary(average_metrics)
+        
+        update_submission_result_and_exit(
+            challenge_pk=self.challenge_pk,
+            phase_pk=self.phase_pk,
+            submission_pk=self.submission_pk,
+            metrics=average_metrics,
+            submission_status="FINISHED",
+        )
+        
+    def close(self, request, context):
+        self._env.close()
+        return evaluation_pb2.Package()                    
+    
+@click.command()
+@click.option("--test-pick", default=False, is_flag=True)
+@click.option("--test-gaze", default=False, is_flag=True)
+@click.option("--test-place", default=False, is_flag=True)
+@click.option("--skip-gaze", default=True, is_flag=True)
+@click.option("--reset-nav", default=False, is_flag=True)
+@click.option("--dry-run", default=False, is_flag=True)
+@click.option("--pick-object", default="cup")
+@click.option("--start-recep", default="table")
+@click.option("--goal-recep", default="chair")
+@click.option(
+    "--cat-map-file", default="projects/real_world_ovmm/configs/example_cat_map.json"
+)
+@click.option("--max-num-steps", default=200)
+@click.option("--visualize-maps", default=False, is_flag=True)
+@click.option("--visualize-grasping", default=False, is_flag=True)
+@click.option(
+    "--debug",
+    default=False,
+    is_flag=True,
+    help="Add pauses for debugging manipulation behavior.",
+)
+def main():
+    try:
+        BODY = os.environ.get("BODY")
+        BODY = BODY.replace("'", '"')
+        BODY = json.loads(BODY)
+        print("BODY", BODY)
+
+        challenge_pk = BODY["challenge_pk"]
+        phase_pk = BODY["phase_pk"]
+        submission_pk = BODY["submission_pk"]
```

Comment:
I think all these are parameters specific to EvalAI and you won't have them in the real-world experiments. I suggest you get rid of the challenge_pk, phase_pk and submission_pk variables..